pkg/history/sqlitestore/store_test.go

```diff
@@ -505,3 +561,160 @@ func newTestDB(tb testing.TB) *Database {
+				{ID: records[1].ID.String(), CreateTime: originTime.Add(time.Millisecond), Payload: []byte("payload-2")},
+				{ID: records[0].ID.String(), CreateTime: originTime, Payload: []byte("payload-1")},
+			},
+		},
+		"from_id": {
+			slice: true,
+			from:  history.Record{ID: records[2].ID.String()},
+			expect: []history.Record{
+				{ID: records[2].ID.String(), CreateTime: originTime.Add(time.Millisecond), Payload: []byte("payload-3")},
+			},
+		},
+		"from_time": {
+			slice: true,
+			from:  history.Record{CreateTime: originTime.Add(time.Millisecond)},
+			expect: []history.Record{
+				{ID: records[1].ID.String(), CreateTime: originTime.Add(time.Millisecond), Payload: []byte("payload-2")},
+				{ID: records[2].ID.String(), CreateTime: originTime.Add(time.Millisecond), Payload: []byte("payload-3")},
+			},
+		},
+		"to_id": {
+			slice: true,
+			to:    history.Record{ID: records[2].ID.String()},
+			expect: []history.Record{
+				{ID: records[0].ID.String(), CreateTime: originTime, Payload: []byte("payload-1")},
+				{ID: records[1].ID.String(), CreateTime: originTime.Add(time.Millisecond), Payload: []byte("payload-2")},
+			},
+		},
+		"to_time": {
+			slice: true,
+			to:    history.Record{CreateTime: originTime.Add(time.Millisecond)},
+			expect: []history.Record{
+				{ID: records[0].ID.String(), CreateTime: originTime, Payload: []byte("payload-1")},
+			},
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			store := db.OpenStore(source)
+			var slice history.Slice = store
+			if tc.slice {
+				slice = slice.Slice(tc.from, tc.to)
+			}
+			buf := make([]history.Record, len(tc.expect)+1)
+			var n int
+			var err error
+			if tc.desc {
+				n, err = slice.ReadDesc(ctx, buf)
+			} else {
+				n, err = slice.Read(ctx, buf)
+			}
+			if err != nil {
+				t.Fatalf("unexpected error: %v", err)
+			}
+			if n != len(tc.expect) {
+				t.Errorf("expected %d records, got %d", len(tc.expect), n)
+			}
+			diff := cmp.Diff(tc.expect, buf[:n])
+			if diff != "" {
+				t.Errorf("data mismatch (-want +got):\n%s", diff)
+			}
+
+			// also grab Len and verify it matches
+			length, err := slice.Len(ctx)
+			if err != nil {
+				t.Errorf("unexpected Len error: %v", err)
+			}
+			if length != len(tc.expect) {
+				t.Errorf("expected Len %d, got %d", len(tc.expect), length)
+			}
+		})
+	}
+
+}
+
+func TestStore_Append(t *testing.T) {
+	db := newTestMemDB(t)
+	ctx := t.Context()
+	source := "test-source"
+
+	store := db.OpenStore(source)
+	record, err := store.Append(ctx, []byte("test-payload"))
+	if err != nil {
+		t.Fatalf("unexpected Append error: %v", err)
+	}
+	diff := cmp.Diff(history.Record{
+		CreateTime: time.Now(),
+		Payload:    []byte("test-payload"),
+	}, record,
+		cmpopts.IgnoreFields(history.Record{}, "ID"),
+		// we can't control exactly what time is used, but it shouldn't vary much from now
+		cmpopts.EquateApproxTime(time.Second),
```

Comment:
Would this get better with synctest?

Reply:
Yes, because synctest has a deterministic clock so we'd know exactly what time to expect.